src/uproot/source/fsspec.py

```diff
@@ -86,7 +86,10 @@ def __enter__(self):
 
     def __exit__(self, exception_type, exception_value, traceback):
         self._fh = None
-        self._executor.shutdown()
+        # TODO: proper cleanup. Shutting down the loop executor causes problems
+        # self._executor.shutdown()
+        if not isinstance(self._executor, uproot.source.futures.LoopExecutor):
+            self._executor.shutdown()
         self._file.__exit__(exception_type, exception_value, traceback)
```

Comment:
I think the file exit should be before the executor shuts down

Reply:
I tried it but it doesn't make a difference.